submissions/kovalov/js-pre-oop/index.js

```diff
@@ -1,45 +1,52 @@
-function createInhabitant(obj) {
-  return new Map(Object.entries(obj));
-}
-
-const dog = createInhabitant({
+const dog = {
   species: 'dog',
   name: 'Patron',
   gender: 'male',
   legs: 4,
   hands: 0,
   saying: 'woof',
-});
+};
 
-const cat = createInhabitant({
+const cat = {
   species: 'cat',
   name: 'Murzyk',
   gender: 'male',
   legs: 4,
   hands: 0,
   saying: 'meow',
-});
+};
 
-const woman = createInhabitant({
+const woman = {
   species: 'human',
   name: 'Anna',
   gender: 'female',
   legs: 2,
   hands: 2,
   saying: 'hello',
-});
+};
 
-const man = createInhabitant({
+const man = {
   species: 'human',
   name: 'Joey',
   gender: 'male',
   legs: 2,
   hands: 2,
   saying: 'How you doin?',
-});
+};
 
-const inhabitants = [dog, cat, woman, man];
+const catWoman = Object.create(cat);
+catWoman.name = 'Cat-woman';
+catWoman.species = 'human';
+catWoman.gender = 'female';
+catWoman.legs = 2;
+catWoman.hands = 2;
+
+const inhabitants = [dog, cat, woman, man, catWoman];
+
+const keys = ['species', 'name', 'gender', 'legs', 'hands', 'saying'];
+
+const details = inhabitants.map((item) => {
+  return keys.map((key) => item[key]);
+});
 
-inhabitants
-  .map((item) => Array.from(item.values()))
-  .forEach((item) => print(item.join('; ')));
+details.forEach((item) => print(item.join('; ')));
```

Comment:
### 1. Arrow functions and chaining

If an arrow's function is a one-liner then you may skip `return` keyword and curly brackets.
If arrow function has a single argument then parentheses can be omitted.

Compare
```
const details = inhabitants.map((item) => {
  return keys.map((key) => item[key]);
});

const details = inhabitants.map(item => keys.map(key => item[key]));
```
Definitely, if you have more nesting levels in your callbacks you will want to split the line into several lines for readability or even split into several independent expressions (just like you've done).
Having no extra keywords like `return` and unnecessary brackets will improve readability.
```
const details =  inhabitants.map((item) => {
  return keys.map((key) => item[key]);
});

const details = inhabitants.map(item =>
  keys.map(key => item[key])
);

// or
const details = inhabitants.map(
  item => keys.map(key => item[key])
);

// or
const details = inhabitants.map(
  item => keys.map(key => item[key]).join("; ") 
  // with `.join` each entry in `details` array represents a ready made presentation string
); // here you may chain `.forEach` and wouldn't need `details` variable
```
Linter will help you to format code for better readability but will unlikely refactor the code to have no `return` statements.

### 2. Naming

Avoid generic, unrelated to code business domain terms and words when naming variables and functions/methods, whenever you implement a specific business logic. Generics are OK when we design a multipurpose functions or library/SDK.

Imagine, your codebase is split into fragments - a method, or function, or declaration are each in their own file.

So reading a specific file must tell the reader what is going on there, provided they know general context of the task.

Variable names
- `dog`, `cat` etc - good, clear what/who an object represents;
- `inhabitants` - good, clearly understandable what to expect to see inside: a collection of world innhabitans;
- `keys` - too generic; we **know** it contains a list of an inhabitant's properties; it would be better if it **told** the reader this, e.g. **inhabitantPropertyNames**;
- `key` - too generic; what is it? an inhabitant's prop name? tell this;
- `item` - too generic; what specifically a callback receives? is it an inhabitant or an inhabitant's details? tell this;

Don't save characters or typing effort on naming. Luckily, IDE offers autoco,pletion once a variable defined.